core/src/packets/icmp/v6/mod.rs

```diff
@@ -252,7 +252,7 @@ impl<E: Ipv6Packet> Packet for Icmpv6<E> {
     /// * [`checksum`] is computed based on the [`pseudo-header`] and the
     /// full packet.
     ///
-    /// [`checksum`]: Icmpv6::checksum
+    /// [`checksum`]: Icmpv6Packet::checksum
```

Comment:
noticed that functions on `Icmpv4` and `Icmpv6` have inconsistent visibility scopes. `msg_type`, `code`, `set_code`, `checksum`, `compute_checksum` should all be pub on Icmpv6. then don't need to fix these links anymore.

Reply:
good. I was going to do this, but wasn't sure why we had the inconsistency.

Reply:
yep not sure how I missed this, but they need to be public so extending `Icmpv6` is possible from outside the crate.